modules/auxiliary/client/telegram/send_message.rb

```diff
@@ -3,54 +3,104 @@
 # Current source: https://github.com/rapid7/metasploit-framework
 ##
 
-require 'uri'
+require 'faraday'
 
 class MetasploitModule < Msf::Auxiliary
   def initialize
     super(
     'Name' => 'Telegram Message Client',
     'Description' => %q{
-            This module will send a Telegram message to given chat ID with the
-            given bot token. Please refer to the module documentation for info
-            on how to retrieve the bot token and corresponding chat ID values.
+        This module can be used to send a specified document and message
+        to multiple users with an optional message and can be used for
+        ethical phishing campaigns. Please refer to the module documentation
+        for info on how to retrieve the bot token and corresponding chat ID
+        values.
         },
-    'Author' => 'Ege Balcı <egebalci[at]pm.me>', # Aka @egeblc of https://pentest.blog
+    'Author' =>
+      [
+        'Ege Balcı <egebalci[at]pm.me>', # Aka @egeblc of https://pentest.blog
+        'Gaurav Purswani' # @pingport80
+      ],
     'License' => MSF_LICENSE,
     )
 
     register_options(
       [
         OptString.new('BOT_TOKEN', [true, 'Telegram BOT token', '']),
-        OptInt.new('CHAT_ID', [true, 'Chat ID for the BOT', '']),
-        OptString.new('MSG', [true, 'Message content', 'New session opened!']),
-        OptEnum.new('FORMATTING', [true, 'Message formating option (Markdown|MarkdownV2|HTML)', 'Markdown', [ 'Markdown', 'MarkdownV2', 'HTML']]),
+        OptString.new('MESSAGE', [false, 'Optional message sent with the document']),
+        OptInt.new('CHAT_ID', [false, 'Chat ID for the BOT', '']),
+        OptPath.new('DOCUMENT', [false, 'The path to the document(binary, video etc)']),
+        OptPath.new('IDFILE', [false, 'File containing chat IDs, one per line'])
       ], self.class
     )
   end
 
   def message
-    datastore['MSG']
+    datastore['MESSAGE']
   end
 
-  def formatting
-    datastore['FORMATTING']
+  def document
+    datastore['DOCUMENT']
   end
 
   def bot_token
     datastore['BOT_TOKEN']
   end
 
-  def run
-    uri = URI("https://api.telegram.org/bot#{bot_token}/sendMessage")
-    params = { chat_id: datastore['CHAT_ID'], parse_mode: formatting, text: message }
```

Comment:
To me this `parse_mode` is important to keep in to allow for sending messages with different formats.

Reply:
yes, I'll add this :+1: